kv/memberlist/memberlist_client_test.go

```diff
@@ -1295,3 +1295,16 @@ type testLogger struct {
 func (l testLogger) Log(keyvals ...interface{}) error {
 	return nil
 }
+
+type providerImpl struct {
```

Comment:
[nit] Which provider? A DNS provider! So I would name it `dnsProviderMock`. "Mock" because it's not a real implementation but just a mock.

```suggestion
type dnsProviderMock struct {
```

Reply:
@pracucci 👍

Reply:
Done.